File: challenge_99/challenge_99.exs
defmodule Exponents do
	def input(file) do
		case File.read(file) do
		  {:ok, body} 		->
		  	body
		  		|> String.split("\n", trim: true)
		  		|> Stream.map(fn b -> String.split(b,",") end)
		  		|> Stream.map(fn [c,d] -> [String.to_integer(c),String.to_integer(d)] end)
					|> Stream.map(fn n -> Enum.at(n,1) * :math.log(Enum.at(n,0)) end) 
					|> Enum.with_index 
					|> Enum.max_by(fn {y,_} -> y end) 
					|> elem(1) 
					|> + 1 
					|> IO.puts
		  {:error,reason} ->
		  	:file.format_error(reason)
		end
	end
end

Exponents.input("challenge_99_exps.txt")

File: challenge_99/challenge_99.rb
base_exp_pairs = []
File.open('challenge_99/challenge_99_exps.txt','r').each_line do |line| 
	nums = line.chomp.split(",").map(&:to_i)
	base_exp_pairs << (nums[1] * Math.log(nums[0])) # Log => instead of exponents
end

p base_exp_pairs.index(base_exp_pairs.max) + 1
